core/Http.php

```diff
@@ -430,7 +439,13 @@ public static function sendHttpRequestBy(
                 fclose($handle);
             } else {
                 $response = @file_get_contents($aUrl, 0, $ctx);
-                if ($response === false) {
+
+                // try to get http status code from response headers
+                if (isset($http_response_header) && preg_match('~^HTTP/(\d\.\d)\s+(\d+)(\s*.*)?~', implode("\n", $http_response_header), $m)) {
+                    $status = (int)$m[2];
+                }
+
+                if (!$status && $response === false) {
```

Comment:
I'm not sure about this -  Imagine if $status == `403` but $response is `false` then shouldn't we throw an exception?